cloud/metadata/gitlab.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+//nolint:revive
+package metadata
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/terramate-io/terramate/cmd/terramate/cli/gitlab"
+)
+
+type GitlabMergeRequest struct {
+	Author *GitlabUser `json:"user,omitempty"`
+
+	DetailedMergeStatus string `json:"detailed_merge_status,omitempty"`
+
+	// Reviewers contains only the latest review per reviewer.
+	Reviewers    []GitlabMergeRequestReviewer `json:"reviewers,omitempty"`
+	Participants []GitlabUser                 `json:"participants,omitempty"`
+	Assignees    []GitlabUser                 `json:"assignees,omitempty"`
+}
+
+type GitlabUser struct {
+	ID        string `json:"id,omitempty"`
+	Username  string `json:"username,omitempty"`
+	Name      string `json:"name,omitempty"`
+	WebURL    string `json:"web_url,omitempty"`
+	AvatarURL string `json:"avatar_url,omitempty"`
+}
+
+type GitlabMergeRequestReviewer struct {
+	User      *GitlabUser `json:"user,omitempty"`
+	CreatedAt *time.Time  `json:"created_at,omitempty"`
+	// State can contain the following values:
+	// * unreviewed - reviewer is assigned but has not reviewed yet
+	// * reviewed - reviewed with comments, but no approval decision yet
+	// * requested_changes - reviewed with requested changes
+	// * approved - reviewed and approved
+	// * unapproved - previous approval was revoked
+	// These cases have been tested, but there may be more potential values.
+	State string `json:"state,omitempty"`
+}
+
+func NewGitlabMergeRequest(inMR *gitlab.MR, inReviewers []gitlab.MRReviewer, inParticipants []gitlab.User) (*GitlabMergeRequest, error) {
+	author, err := NewGitlabUser(&inMR.Author)
+	if err != nil {
+		return nil, err
+	}
+
+	reviewers := make([]GitlabMergeRequestReviewer, 0, len(inReviewers))
+	for _, e := range inReviewers {
+		p, err := NewGitlabMergeRequestReviewer(&e)
+		if err != nil {
+			return nil, err
+		}
+		reviewers = append(reviewers, *p)
+	}
+
+	participants := make([]GitlabUser, 0, len(inParticipants))
+	for _, e := range inParticipants {
+		p, err := NewGitlabUser(&e)
+		if err != nil {
+			return nil, err
+		}
+		participants = append(participants, *p)
+	}
+
+	assignees := make([]GitlabUser, 0, len(inMR.Assignees))
+	for _, e := range inMR.Assignees {
+		p, err := NewGitlabUser(&e)
+		if err != nil {
+			return nil, err
+		}
+		assignees = append(assignees, *p)
+	}
+
+	return &GitlabMergeRequest{
+		Author:              author,
+		DetailedMergeStatus: inMR.DetailedMergeStatus,
+		Reviewers:           reviewers,
+		Participants:        participants,
+		Assignees:           assignees,
+	}, nil
+}
+
+func NewGitlabUser(in *gitlab.User) (*GitlabUser, error) {
+	return &GitlabUser{
+		ID:        strconv.Itoa(in.ID),
+		Username:  in.Username,
+		Name:      in.Name,
+		WebURL:    in.WebURL,
+		AvatarURL: in.AvatarURL,
+	}, nil
+}
+
+func NewGitlabMergeRequestReviewer(in *gitlab.MRReviewer) (*GitlabMergeRequestReviewer, error) {
```

Comment:
Same here as it `NewGitlabUser` neither.
```suggestion
func NewGitlabMergeRequestReviewer(in *gitlab.MRReviewer) *GitlabMergeRequestReviewer {
```